server/src/server.ts

```diff
@@ -120,7 +128,17 @@ app.get(
     failureMessage: true,
   }),
   function (req, res) {
-    res.redirect("/");
+    const { redirectUrl } = JSON.parse(
+      Buffer.from(
+        decodeURIComponent(req.query.state as string),
+        "base64"
+      ).toString("utf-8")
+    );
```

Comment:
any chance this blows up if you feed it garbage?

Reply:
Yeah good catch. JSON.parse can throw an exception. I've updated to wrap them in try/catch just in case.